lib/bundle/utils.rb

```diff
@@ -16,16 +16,10 @@ def self.system(cmd, *args)
     success
   end
 
-  def self.brew_installed?
-    @brew ||= Kernel.system("brew --version >/dev/null 2>&1")
-  end
-
   def self.cask_installed?
-    @cask ||= Kernel.system("brew command cask >/dev/null 2>&1")
-  end
-
-  def self.services_installed?
-    @services ||= Kernel.system("brew command services >/dev/null 2>&1")
+    @cask ||= begin
+      which("brew-cask") || which("brew-cask.rb")
```

Comment:
OK, your call.

(To be honest, the only thing I worry about is that after seeing these codes, other people start using private API in their own external commands, which could be easily broken)